sdk/tests/core/test_az_url_encode.c

```diff
@@ -84,6 +85,42 @@ static void test_url_encode_basic(void** state)
     assert_true(az_span_is_content_equal(
         AZ_SPAN_FROM_BUFFER(buf20), AZ_SPAN_FROM_STR("aBc%2Fg*************")));
   }
+  {
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("aBc/g"));
+    assert_int_equal(url_length, sizeof("aBc%2Fg") - 1);
+  }
+  {
+    // Empty span
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_NULL);
+    assert_int_equal(url_length, 0);
+    url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR(""));
+    assert_int_equal(url_length, 0);
+  }
+  {
+    // Nothing gets encoded
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("123"));
+    assert_int_equal(url_length, 3);
+  }
+  {
+    // first last encoded
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR(" 123 "));
+    assert_int_equal(url_length, 9);
+  }
+  {
+    // every character is encoded
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("   "));
+    assert_int_equal(url_length, 9);
+  }
+  {
+    // % character, which in itself is encoded as %25
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("%"));
+    assert_int_equal(url_length, 3);
+  }
+  {
+    // a single length span with the \0 NULL character
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("\0"));
+    assert_int_equal(url_length, 3);
+  }
```

Comment:
nit: Move all these tests as a group to the end of the method, after this line:
https://github.com/Azure/azure-sdk-for-c/blob/ce4d71bbd0f8a49ab79a47a5f23222bb659773c6/sdk/tests/core/test_az_url_encode.c#L240-L243

Right now they are right in the middle between a bunch of `_az_span_url_encode` tests